packages/mc-scripts/src/commands/config:sync.js

```diff
@@ -0,0 +1,136 @@
+  return customAppLink;
+};
+
+const configSync = async () => {
+  const applicationConfig = processConfig();
+  const { data: localCustomAppData } = applicationConfig;
+  const { mcApiUrl } = applicationConfig.env;
+
+  if (!credentialsStorage.isSessionValid(mcApiUrl)) {
+    throw new Error(
+      `You don't have a valid session for the ${mcApiUrl} environment. Please, run the "mc-scripts login" command to authenticate yourself.`
+    );
+  }
+
+  const token = credentialsStorage.getToken(mcApiUrl);
+  const fetchedCustomApplication = await fetchCustomApplication({
+    mcApiUrl,
+    token,
+    entryPointUriPath: localCustomAppData.entryPointUriPath,
+  });
+
+  if (!fetchedCustomApplication) {
+    const userOrganizations = await fetchUserOrganizations({ mcApiUrl, token });
+
+    let organizationId, organizationName;
+    if (userOrganizations.total > 1) {
+      const organizationChoices = userOrganizations.results.map(
+        (organization) => ({
+          title: organization.name,
+          value: organization.id,
+        })
+      );
+
+      organizationId = await prompts({
+        type: 'select',
+        name: 'organizationId',
+        message: 'Select Organization',
+        choices: organizationChoices,
+        initial: 0,
+      })['organizationId'];
+
+      organizationName = organizationChoices.find(
+        ({ value }) => value === organizationId
+      ).title;
+    } else {
+      organizationId = userOrganizations.results[0].id;
+      organizationName = userOrganizations.results[0].name;
+    }
+
+    const { confirmation } = await prompts({
+      type: 'text',
+      name: 'confirmation',
+      message: `You are about to create a new Custom Application in the "${organizationName}" organization for the ${mcApiUrl} environment. Are you sure you want to proceed?`,
+      initial: 'yes',
+    });
+    if (confirmation.toLowerCase().charAt(0) !== 'y') {
+      console.log(chalk.red('Aborted.'));
+    } else {
+      const createdCustomApplication = await createCustomApplication({
+        mcApiUrl,
+        token,
+        organizationId,
+        data: omit(localCustomAppData, ['id']),
+      });
+      // update applicationID in the custom-application-config file
+      updateApplicationIdInCustomApplicationConfig(createdCustomApplication.id);
+      const customAppLink = getMcUrlLink(
+        mcApiUrl,
+        organizationId,
+        createdCustomApplication.id
+      );
+      console.log(
+        chalk.green(
+          `Custom Application created.\nThe "applicationId" in your local Custom Application config file has been updated with the application ID.\nYou can see the Custom Application data in the Merchant Center at ${customAppLink}.`
+        )
+      );
+    }
+    return;
+  }
+  // update applicationID in the custom-application-config file
+  updateApplicationIdInCustomApplicationConfig(
+    fetchedCustomApplication.application.id
+  );
```

Comment:
Is it actually possible for the ID of a Custom Application to change after its creation?

Reply:
Not to my knowledge. It's a Prisma generated cuid.

Reply:
No. But, it is possible that the user created the custom app on the merchant center and not with the cli.

Reply:
The function is to update the local custom-application-config with the applicationId.

Reply:
Speaking of the `applicationId`, we need to consider this use case: the config might use an env placeholder for the application id, e.g. `${env:APPLICATION_ID}`.

In this case, we can't really replace the value.

I suppose we should only do that when we create a new Custom App. When updating, we shouldn't rewrite the value.

@Rhotimee Could you adjust the logic? Thanks

Reply:
sure 👍🏽